### PR TITLE
[FIX] Fix weight dying in BERTModel.decoder for BERT pre-training

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
The decoder for masked language model task in BERTModel is not correctly tied with the word embedding similar to https://github.com/dmlc/gluon-nlp/pull/413 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- Does this affect other downstream fine-tuning tasks? No, it affects the decoder which is only relevant to the masked language model task during pre-training. 
- Does this affect the pre-trained models? No, the converted BERT model from TF can still be loaded correctly.

### CODE REVIEW

**Comment by mli** [comment]
Job PR-500/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-500/4/index.html

File: scripts/bert/convert_tf_model.py [hunk 1]
```diff
@@ -29,12 +29,16 @@
 
 parser = argparse.ArgumentParser(description='Conversion script for Tensorflow BERT model')
 parser.add_argument('--model', type=str, default='bert_12_768_12',
-                    help='BERT model name. options are bert_12_768_12 and bert_24_1024_16')
-parser.add_argument('--tf_checkpoint_dir', type=str, required=True,
+                    help='BERT model name. options are bert_12_768_12 and bert_24_1024_16.'
+                         'Default is bert_12_768_12')
+parser.add_argument('--tf_checkpoint_dir', type=str,
```
Comment (cgraywang) [inline scripts/bert/convert_tf_model.py]: What's the reason for us to provide the tensorflow convertion script? I think it might be better to remove this script, since we have provided our own parameters.
